packages/nimble-components/src/wafer-map/specs/features/rendering.md

```diff
@@ -210,8 +211,69 @@ The current expectation is for a singular wafer component to be displayed on the
 
 ### Alternative Data Structures and Interfaces
 
+The alternative to using Apache Arrow tables is an in-house solution:
+
+```TS
+class WaferData {
+    // the x coordinates of each column of dies
+    dieColIndexArray: Int32Array;
+    // the lengths of each row of dies
+    rowLengthsArray: Int32Array;
+    // the y coordinates of each die as a matrix row by row
+    dieRowIndexLayer: Int32Array;
+    // the value of each die as a matrix row by row
+    dieValuesLayer: Float64Array;
+    // the highlight  approach is still undecided, we have two options:
+    // the highlight state of each die as a matrix; user will have to pre-calculate tags into highlighted conditions.
+    dieHighlightsLayer: Int8Array;
+    // a 32 bitset array of tags for each die; aligns more closely to the existing public api but limits users to 32 tags.
+    dieHighlightsLayer: Int32Array;
+    // metadata array for each die; it will not be sent to the worker
+    metadata : unknown[]
+}
+```
+
+Using TypedArrays has the benefit of direct transfer to web workers without structured cloning of the object by transferring the arrayBuffers and reconstructing the object. Other benefits of typedArrays include the low access time when iterating over the values, more memory efficiency and faster direct access to metadata layers values. The previous inputs can be adapted to this new structure to maintain backwards compatibility.
+
+This API will have [optimized byte-array interop from Blazor](https://learn.microsoft.com/en-us/dotnet/core/compatibility/aspnet-core/6.0/byte-array-interop) and should be supported by Angular as a [vanilla javascript feature](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/ArrayBuffer).
```

Comment:
The document describes how we'd handle Angular/Blazor integration for this TypedArrays alternative but not for the recommended Apache Arrow approach. Could you add an equivalent paragraph to the section above?